vendor/github.com/prometheus/tsdb/db.go

```diff
@@ -382,26 +382,41 @@ func (db *DB) compact() (changes bool, err error) {
 			return changes, nil
 		default:
 		}
-		// The head has a compactable range if 1.5 level 0 ranges are between the oldest
-		// and newest timestamp. The 0.5 acts as a buffer of the appendable window.
-		if db.head.MaxTime()-db.head.MinTime() <= db.opts.BlockRanges[0]/2*3 {
+
+		minHead := db.head.MinTime()
+		maxHead := db.head.MaxTime()
+		maxBlock := minHead + db.opts.BlockRanges[0]
```

Comment:
This will cause blocks to cross the virtual block boundaries aligned at each `db.opts.BlockRanges[0]`. This discards a very central paradigm that lot of TSDB assumes as an invariant.

I _think_ the compactor today might just skip such blocks and then later, at a higher-level compaction catch those blocks. 
But I'm by no means sure about that. TSDB certainly wasn't build with this in mind and it will almost guaranteed cause trouble somewhere.